DOCS/man/input.rst

```diff
@@ -1861,12 +1861,16 @@ Property list
     looks better for display purposes. Use the ``path`` property to get an
     unmodified filename.)
 
-    This has a sub-property:
+    This has two sub-properties:
 
     ``filename/no-ext``
         Like the ``filename`` property, but if the text contains a ``.``, strip
         all text after the last ``.``. Usually this removes the file extension.
 
+    ``filename/ext``
+        Like the ``filename/no-ext`` property, but returns the text after the
+        last ``.`` instead.
```

Comment:
If a file extension doesn't exist then I'd expect it to return an empty string. Not the filename.

Reply:
Note that this doesn't apply to `filename/no-ext` because if the file doesn't have an extension, then there's nothing to strip. So the user gets what he asked for already, a filename without extension.

But if the user asks for the extensions specifically, it doesn't make sense to return the filename.

Reply:
Yeah true, that was the original iteration of this but I wasn't sure which was better. I'll just revert it to that later

Reply:
The current behavior seems to be to return that it is unavailable as opposed to an empty string. I have no real strong opinion on what's better but I thought I'd mention it.

Reply:
I suggested empty string because it can often naturally fit in with the program's logic avoiding unnecessary special casing.

But I'm not 100% sure where this will be used/is likely to be used so I don't have any strong opinion either.

But that leads me to the question: is there a concrete usecase for this or is it being added just because? (It's simple enough that I don't care either ways, but I'm curious).

Reply:
An empty string is probably preferable now that I think about it some more.

>is there a concrete usecase for this or is it being added just because?

Yeah no idea. I don't really mind either way though.

Reply:
> is there a concrete usecase for this or is it being added just because?

It is to simplify conditions like `profile-cond=string.lower(string.match(get("filename", ""), ".+%.(.+)$")) == "gif"`

Technically, there's no reason for the `no-ext` case to be there as well, since it can just be done with regex. And this is helpful for replicating the deprecated `[extension.gif]` auto profiles, since they'll eventually be removed.

>An empty string is probably preferable now that I think about it some more.

I'll change it to that

Reply:
Why not use `video_format == 'gif'`?

Reply:
Because it doesn't necessarily output the file extension, but the codec instead. I used `gif` as an example, but if it's a h264 mkv file then `video-format` will be h264, not mkv. Either way, if this is bloat then I'll just close it and keep using string.match. We could also remove the no-ext sub property.

Reply:
For mkv you can check `current-demuxer`.

Reply:
And for mp4? `mov,mp4,m4a,3gp,3g2,mj2` is my favorite file extension. Either way string.match works just fine, I was only interested in making it possible to do this without regex matching.

Reply:
> if this is bloat then I'll just close it

I think it's fine to add it tbh. If the user is fine relying on extensions (which might not be correct) then that's the user's choice.

Reply:
I don't know why you would need to match the mp4 container instead of checking `video-codec`, but you can shorten the condition `profile-cond=filename:lower():find('%.mp4$')`